daemon/cluster/executor/container/controller.go

```diff
@@ -52,7 +55,11 @@ func newController(b executorpkg.Backend, task *api.Task, secrets exec.SecretGet
 	}, nil
 }
 
+// Task returns the controller's task.
 func (r *controller) Task() (*api.Task, error) {
+	// concurrency safe
+	r.taskMu.RLock()
+	defer r.taskMu.RUnlock()
```

Comment:
Also, is this really safe? Since this is returning a pointer to the task the caller of this function can't access this returned object concurrently.